downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,759 @@
+ 
+type E3 distinct error;
+ 
+type E4 distinct error;
+ 
+type E5 E1 & E2; // not allowed to access outside the module
+ 
+type E6 E3 & E4; // not allowed to access outside the module
+```
+
+- Fixed a bug that resulted in a compilation error not being logged for invalid usage of an on-conflict clause in a query with inner queries.
+
+```ballerina
+type Token record {|
+    readonly int idx;
+    string value;
+|};
+
+type TokenTable table<Token> key(idx);
+
+public function main() {
+    Token[] tbl = from Token i in (table key(idx) from var j in [1, 2]
+                        select {
+                            idx: j,
+                            value: "A"
+                        })
+                    select {
+                        idx: i.idx,
+                        value: "A"
+                    }
+                    on conflict error("Duplicate Key"); // Now an error
+}
+```
+
+- A bug that resulted in a compilation error not being logged when a spread field is used in a table member to specify a field that is part of the key sequence has been fixed. For every field name in the key sequence of a table, a mapping constructor member must have a specific field corresponding to the field.
+
+```ballerina
+type Employee record {
+    readonly int id;
+    string name;
+    string dept;
+};
+
+public function main() {
+    Employee employee = {
+        id: 1234,
+        name: "Amy Wilson",
+        dept: "legal"
+    };
+
+    table<Employee> key(id) employees = table [
+        // Results in an error now.
+        {...employee}
+    ];
+}
+```
+
+To view bug fixes, see the [GitHub milestone for 2201.2.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%22Ballerina+2201.2.0%22).
+
+### Runtime updates
+
+#### New features
+
+##### Strand dump
+
+Allows getting the status of strands and strand groups during the execution of a Ballerina program.
+
+This can be used to troubleshoot runtime errors. The Ballerina runtime will emit the strand dump to the standard output stream in the text format if it receives a `SIGTRAP` signal (`SIGTRAP` is not available on Windows).
+
+E.g., if the PID of the running Ballerina program is `$PID`, you can get the strand dump by executing either `kill -SIGTRAP $PID` or `kill -5 $PID` command.
+
+##### `StopHandler` Object
+
+Allows registering a function that will be called during a graceful shutdown.
+
+A call to `onGracefulStop` will result in one call to the handler function that was passed as an argument; the handler functions will be called after calling `gracefulStop` on all registered listeners, in the reverse order of the corresponding calls to `onGracefulStop`.
+
+E.g., a `foo` function can be called during the graceful shutdown by registering it as follows.
+
+`runtime:onGracefulStop(foo);`
+
+#### Improvements
+
+
+#### Type-reference type support at runtime
```

Comment:
This should be a L5 heading ? @praneesha

Reply:
Fixed.